elastic/kibana
# [Fleet] Remove aliases from index_template when updating an existing index template

## Description

Resolve https://github.com/elastic/kibana/issues/90984

Ensure we aliases from index_template when updating an existing index template

## Context 

When updating an index template we put new component template before updating the index template, this cause validation error in the case where the index template contains the `aliases`  property. That PR ensure the aliases property is removed before.

## How to test

Start with a Kibana 7.10 setup fleet and endpoint than update to that PR you should be able to upgrade and go to fleet and the index_template for `logs-endpoint*` should not contains aliases

Locally I tested it like this (without the fix I got the issue) 


First run kibana from the `7.10` branch,
* run Elastic with data in a persistent place, like this `yarn es snapshot --license trial --password changeme -E xpack.security.authc.api_key.enabled=true -E path.data=/tmp/es-data` 
* Run kibana  (do not forget the bootstrap, ...)
* Setup fleet and install endpoint

Than checkout that PR
* run Elastic with data in a persistent place, like this `yarn es snapshot --license trial --password changeme -E xpack.security.authc.api_key.enabled=true -E path.data=/tmp/es-data` 
* Run kibana  (do not forget the bootstrap, ...)
* Go to fleet it should be fixed.
